src/main/java/htmlpublisher/HtmlPublisherTarget.java

```diff
@@ -114,16 +122,18 @@ public String getReportTitles() {
      * @param alwaysLinkToLastBuild If true, the job action will refer the latest build.
      *      Otherwise, the latest successful one will be referenced
      * @param allowMissing If true, blocks the build failure if the report is missing
+     * @param onlyCreateReportWithDifferentName If true, 2 calls to publish HTML reports with the same name will create 2 links on build page
      * @since 1.4
      */
     @DataBoundConstructor
-    public HtmlPublisherTarget(String reportName, String reportDir, String reportFiles, boolean keepAll, boolean alwaysLinkToLastBuild, boolean allowMissing) {
+    public HtmlPublisherTarget(String reportName, String reportDir, String reportFiles, boolean keepAll, boolean alwaysLinkToLastBuild, boolean allowMissing, boolean onlyCreateReportWithDifferentName) {
```

Comment:
For the newer fields being added, as they aren't fields that can't otherwise have a default, we are now using `@DataBoundSetter` on the field itself rather than updating the constructors. Note this means you'll need a `setOnlyCreateReportWithDifferentName` method defined with the `@DataBoundSetter` annotation on it. See `escapeUnderscores` for a guide.

Reply:
corrected